src/modules/roc_core/target_linux/roc_core/timer.cpp

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2020 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <errno.h>
+#include <unistd.h>
+
+#include "roc_core/errno_to_str.h"
+#include "roc_core/panic.h"
+#include "roc_core/timer.h"
+
+namespace roc {
+namespace core {
+
+#if defined(CLOCK_MONOTONIC)
+#define CLOCK_TYPE CLOCK_MONOTONIC
+#else
+#define CLOCK_TYPE CLOCK_REALTIME
+#endif
+
+Timer::Timer()
+    : timerfd_(timerfd_create(CLOCK_TYPE, 0))
+    , deadline_(0) {
+    if (timerfd_ == -1) {
+        roc_panic("timer: timerfd_create(CLOCK_TYPE, 0): %s", errno_to_str().c_str());
+    }
+}
+
+Timer::~Timer() {
+    close(timerfd_);
```

Comment:
Nitpick: we should check for errors here.

Reply:
ok